t/deadlock.t

```diff
@@ -0,0 +1,152 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+use Test::Nginx::Socket::Lua;
+
+plan tests => repeat_each() * (blocks() * 11);
+
+$ENV{TEST_NGINX_HTML_DIR} ||= html_dir();
+
+master_on();
+workers(3);
+run_tests();
+
+__DATA__
+
+=== TEST 1: posting events and handling events, broadcast
+--- http_config
+    lua_package_path "../lua-resty-core/lib/?.lua;lualib/?/init.lua;lualib/?.lua;;";
+    lua_shared_dict dict 1m;
+    init_by_lua_block {
+        require("ngx.process").enable_privileged_agent(4096)
+    }
+    init_worker_by_lua_block {
+        local opts = {
+            broker_id = 2,
+            listening = "unix:$TEST_NGINX_HTML_DIR/nginx.sock",
+        }
+
+        local ev = require("resty.events").new(opts)
+        if not ev then
+            ngx.log(ngx.ERR, "failed to new events")
+        end
+
+        local ok, err = ev:init_worker()
+        if not ok then
+            ngx.log(ngx.ERR, "failed to init_worker events: ", err)
+        end
+
+        local semaphore = require("ngx.semaphore")
+        local sema = semaphore.new(1)
+
+        local id = ngx.worker.id()
+
+        if id == 2 then
+            ngx.shared.dict:set("broker-pid", ngx.worker.pid())
+        end
+
+        if id == 0 or id == 1 then
```

Comment:
How about the `id == 2`? The broker itself also receive the event.

Reply:
@outsinre, broker runs on ´id=2`, see:
https://github.com/Kong/lua-resty-events/pull/47/files#diff-f8f62b1b0529d11f96be1a1f25b7f11df85ce86fb864528bd0d623f14bf22278R23

Reply:
Yes, worker 0 and worker 1 will output debug logs for testing.

Reply:
I just wonder if we should test broker's output as well.

Since broker is down in this test case, it is OK without it.